scripts/ci/runners/sync_authors.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import re
+
+import requests
+import toml
+
+AUTHORS = 'https://raw.githubusercontent.com/apache/airflow-ci-infra/main/authors.toml'
+
+
+req = requests.get(AUTHORS)
+author_list = toml.loads(req.text)
+
+author_set = set()
+for membership in author_list:
+    author_set.update([author for author in author_list[membership]])
+
+authors = ''
+for author in sorted(author_set):
+    authors += f'            "{author}",\n'
+
+authors = authors[:-2]
+
+with open('ci.yml') as handle:
+    new_ci = re.sub(
```

Comment:
https://pypi.org/project/ruamel.yaml/ might be a slightly more maintable way than a regex. Though we'd still need to regex the value

Reply:
Updating the yaml with that does not preserve the contents of the file well, even with `RoundTripLoader/Dumper`. Most comments are stripped, and some parts are rewritten such that I would not be confident it would even work.

Reply:
eg

```
-      DEFAULT_CONSTRAINTS_BRANCH: ${{ needs.build-info.outputs.default-constraints-branch }}
+      DEFAULT_CONSTRAINTS_BRANCH: ${{ needs.build-info.outputs.default-constraints-branch
+        }}
```

Reply:
That particular breakage I've fixed — it wants a `width=180` or similar large number in order to not wrap.

Reply:
Happy to merge this as is if you don't think ruamel.yaml is worth it (it may not be)

Reply:
I've tried all the options I can find and still can't get it to roundtrip without stripping out many of the comments. Doesn't seem like a good idea. I could document the regexp better though.

Reply:
(done, sort-of)